apple/Sources/FerrostarCore/NavigationState.swift

```diff
@@ -20,38 +20,4 @@ public struct NavigationState: Hashable {
         self.routeGeometry = routeGeometry
         self.isCalculatingNewRoute = isCalculatingNewRoute
     }
-
-    public var currentProgress: TripProgress? {
```

Comment:
At the callsite, we were using these methods like this:
```
              if case .navigating = navigationState?.tripState,
                        let progress = navigationState?.currentProgress
```

It was a bit of a shame to not be able to take advantage of the type information  — if it's a `.navigating` than we know `progress` isn't nil. Now with the structure we can capture that information and avoid the really verbose destructured tuple assignment.